src/stm32f411/i2c.cpp

```diff
@@ -23,85 +23,91 @@
 #include <libhal-arm-mcu/stm32f411/output_pin.hpp>
 #include <libhal-arm-mcu/stm32f411/pin.hpp>
 #include <libhal-util/static_callable.hpp>
+#include <libhal/i2c.hpp>
 #include <libhal/units.hpp>
 
 #include "libhal-arm-mcu/interrupt.hpp"
 #include "power.hpp"
 
 namespace hal::stm32f411 {
-i2c::i2c(std::uint8_t p_bus_number,
-         i2c::settings const& p_settings,
-         hal::io_waiter& p_waiter)
+i2c_manager_impl::i2c_manager_impl(peripheral select)
+{
+  m_port = select;
+  power(m_port).on();
+}
```

Comment:
```suggestion
i2c_manager_impl::i2c_manager_impl(peripheral p_select): m_port(p_select)
{
  power(m_port).on();
}
```

Prefer to use the initializer list over the constructor body to give everything a defined state before running the body.